src/server/server.ts

```diff
@@ -576,7 +577,84 @@ namespace ts.server {
         }
     }
 
+    function extractWatchDirectoryCacheKey(path: string, currentDriveKey: string) {
+        path = normalizeSlashes(path);
+        if (isUNCPath(path)) {
+            // UNC path: extract server name
+            // //server/location
+            //         ^ <- from 0 to this position
+            const firstSlash = path.indexOf(directorySeparator, 2);

```

Comment:
Any reason this might ever get a UNC path without the next separator (e.g. just "//someshare")? For safety, might want to check for indexOf returning -1 and return undefined.